subsys/nrf_security/src/drivers/cracen/silexpk/src/montgomery.c

```diff
@@ -34,11 +34,12 @@ struct sx_pk_acq_req sx_async_x25519_ptmult_go(const struct sx_x25519_op *k,
 	sx_wrpkmem(inputs.p.addr, pt->bytes, SX_X25519_OP_SZ);
 	sx_wrpkmem(inputs.k.addr, k->bytes, SX_X25519_OP_SZ);
 	/* clamp the scalar */
-	inputs.k.addr[31] |= 1 << 6;
-	inputs.k.addr[31] &= 0x7f;
-	inputs.k.addr[0] &= 0xF8;
+	sx_wrpkmem_byte(&inputs.k.addr[31], inputs.k.addr[31] | 1 << 6);
+	sx_wrpkmem_byte(&inputs.k.addr[31], inputs.k.addr[31] & 0x7f);
```

Comment:
```suggestion
	sx_wrpkmem_byte(&inputs.k.addr[31], (inputs.k.addr[31] | 1 << 6) & 0x7f);
```
? Avoids reading back and stuff, I mean I don't know if the compiler already optimized this out.